sui_core/src/safe_client.rs

```diff
@@ -263,8 +265,52 @@ where
     /// Handle Batch information requests for this authority.
     async fn handle_batch_streaming(
         &self,
-        _request: BatchInfoRequest,
+        request: BatchInfoRequest,
     ) -> Result<Receiver<Result<BatchInfoResponseItem, SuiError>>, io::Error> {
-        todo!()
+        let (mut tx_output, tr_output) = channel(BUFFER_SIZE);
+
+        let mut batch_info_items = self
+            .authority_client
+            .handle_batch_streaming(request)
+            .await?;
+
+        for next_data in batch_info_items.next().await {
+            match next_data {
+                Ok(batch_info_response_item) => {
+                    // do security checks
+                    match batch_info_response_item.clone() {
+                        BatchInfoResponseItem(UpdateItem::Batch(signed_batch)) => {
+                            // check signature of batch
+                            let result = signed_batch.signature.check(&signed_batch.signature, signed_batch.authority);
```

Comment:
Lets add a bunch of todos here for all the other checks:
* The signature should be valid over the set of transactions that the batch contains.
* And also over the hash of the previous batch.
* And the sequence numbers of trasnactions enclosed need to be correct and the size matching?

Reply:
I was going to do these now but these now but realized that this may need more discussion. I've added in todos here.